align_slice.py

```diff
@@ -28,3 +28,27 @@ def align_and_slice(s, dimension='t2', convwidth=500, threshold=0.05, fl=None):
     if fl is not None:
         fl.pop_marker()
     return s[dimension:slices]
+def correlation_align(s,avg,convwidth=0,dimension='t2',fl=None):
```

Comment:
this seems to just be an averaged apodization...I am certain I'm wrong but if not apodization then what?

Reply:
I don't know what you mean

Reply:
for clarification purposes I guess I am asking what the purpose of this function is? what does it add to the overall processing?

Reply:
Since the frequency is drifting, we want to frequency shift each transient slightly so that it's lined up with the average peak.

Inspired by the recent Keller + Maly paper, I try to do this using the cross-correlation -- specifically, I find what frequency the cross-correlation is maximized at, which tells me how much I need to shift the peak over to get it aligned.

See [this page](https://en.wikipedia.org/wiki/Cross-correlation) for more info on the cross-correlation.

Reply:
actually better, please do give [this page](https://www.mathworks.com/help/signal/ug/align-signals-using-cross-correlation.html) a careful look, as well, and see if you agree with the way that this is done.

Reply:
makes much more sense line by line! will talk in indiv meeting at 12 on this!